gax/src/main/java/com/google/api/gax/batching/v2/BatcherImpl.java

```diff
@@ -81,27 +88,32 @@ public BatcherImpl(
       BatchingDescriptor<ElementT, ElementResultT, RequestT, ResponseT> batchingDescriptor,
       UnaryCallable<RequestT, ResponseT> unaryCallable,
       RequestT prototype,
-      BatchingSettings batchingSettings) {
+      BatchingSettings batchingSettings,
+      ScheduledExecutorService executor) {
+
     this.batchingDescriptor =
         Preconditions.checkNotNull(batchingDescriptor, "batching descriptor cannot be null");
     this.unaryCallable = Preconditions.checkNotNull(unaryCallable, "callable cannot be null");
     this.prototype = Preconditions.checkNotNull(prototype, "request prototype cannot be null");
     this.batchingSettings =
         Preconditions.checkNotNull(batchingSettings, "batching setting cannot be null");
+    Preconditions.checkNotNull(executor, "executor cannot be null");
+    currentOpenBatch = new Batch<>(prototype, batchingDescriptor, batchingSettings);
+
+    long delay = batchingSettings.getDelayThreshold().toMillis();
+    scheduledFuture = new PushCurrentBatchRunnable<>(executor, this, delay).self;
```

Comment:
I think it would be better to schedule the runnable here and then set the future on it.

Reply:
As of the moment, I could think of one way to achieve this. Wanted to check if this looks ok to you
```java
class BatcherImpl {
   
  BatcheImpl(...) {
        PushCurrentBatchRunnable runnable = new PushCurrentBatchRunnable<>(this);
    scheduledFuture =
        executor.scheduleWithFixedDelay(runnable, delay, delay, TimeUnit.MILLISECONDS);
    runnable.scheduledFuture = scheduledFuture;
  }

 ...
 ...

  static class PushCurrentBatchRunnable<> implements Runnable {
    private ScheduledFuture<?> scheduledFuture;
    ...
  }
}
```